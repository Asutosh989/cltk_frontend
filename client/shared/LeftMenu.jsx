import Menu from 'material-ui/Menu';
import MenuItem from 'material-ui/MenuItem';
import Divider from 'material-ui/Divider';
import Drawer from 'material-ui/Drawer';
import baseTheme from 'material-ui/styles/baseThemes/lightBaseTheme';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import injectTapEventPlugin from 'react-tap-event-plugin';

injectTapEventPlugin();

LeftMenu = React.createClass({


    propTypes: {
        open: React.PropTypes.bool.isRequired,
        closeLeftMenu: React.PropTypes.func.isRequired,
    },

    getChildContext() {
        return {muiTheme: getMuiTheme(baseTheme)};
    },

    childContextTypes: {
        muiTheme: React.PropTypes.object.isRequired,
    },


	  mixins: [ReactMeteorData],

	  getMeteorData(){
	    var query = {};

			return {
				currentUser: Meteor.users.findOne({_id: Meteor.userId()})
			}

		},

    scrollToAbout(e){
        $("html, body").animate({scrollTop: $('#get-started').height() - 100}, 300);

        this.props.closeLeftMenu();
        e.preventDefault();
    },

    render(){
        var userIsLoggedIn = this.data.currentUser ? true : false;
				var username = "";

				if(userIsLoggedIn){
						if(this.data.currentUser.emails.length){
							username = this.data.currentUser.emails[0].address;
						}

				}


        return (
            <div>
                <Drawer
                    open={this.props.open}
                    docked={false}
                    onRequestChange={open => this.setState({open})}
                    className="md-sidenav-left"
                >
                    <div className="sidenav-top">
                      {userIsLoggedIn ?
                          <div>
                            <div className="user-image paper-shadow">
                              <img src="/images/default_user.jpg"/>
                            </div>
                          </div>
                          : ""
                      }
                      <span className="user-fullname">
												{username}
                      </span>


                    </div>
                    <MenuItem
                        href="/"
                        primaryText="Home"
                        onClick={this.props.closeLeftMenu}
                    />
                    <MenuItem
                        href="/browse"
                        primaryText="Browse"
                        onClick={this.props.closeLeftMenu}
                    />
                    <MenuItem
                          href="/about"
                          primaryText="About"
                          onClick={this.props.closeLeftMenu}
                      />
                    <MenuItem
                          href="//cltk.org"
                          primaryText="CLTK.org"
                          onClick={this.props.closeLeftMenu}
													target="_blank"
                      />
                    <MenuItem
                          href="//github.com/cltk"
                          primaryText="GitHub"
                          onClick={this.props.closeLeftMenu}
													target="_blank"
                      />

                    <Divider />

                    {userIsLoggedIn ?
                      <div>
                        <MenuItem
                            href="/profile"
                            primaryText="Profile"
                            target="_blank"
                            onClick={this.props.closeLeftMenu}
                        />
                        <MenuItem
                            href="/sign-out"
                            primaryText="Sign out"
                            target="_blank"
                            onClick={this.props.closeLeftMenu}
                        />
                    </div>
                    :
                    <MenuItem
                          href="/sign-in"
                          primaryText="Sign in"
                          target="_blank"
                          onClick={this.props.closeLeftMenu}
                      />
                    }
                </Drawer>
            </div>
        );
    }
});
