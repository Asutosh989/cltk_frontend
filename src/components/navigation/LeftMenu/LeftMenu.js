import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Divider from 'material-ui/Divider';
import Drawer from 'material-ui/Drawer';

import { toggleLeftMenu } from '../../../actions/leftMenu';
import MenuItem from '../MenuItem';
import LeftMenuHead from '../LeftMenuHead';

// actions
import { logout, toggleAuthModal } from '../../../modules/auth/actions';
import { logoutUser } from '../../../lib/auth';

// lib
import getCurrentProjectHostname from '../../../lib/getCurrentProjectHostname';

import './LeftMenu.css';


class LeftMenu extends React.Component {

	renderMenuItems() {
		const { closeLeftMenu } = this.props;

		return (
			<div>
				<MenuItem
					to="/collections"
					onClick={closeLeftMenu}
				>
					Collections
				</MenuItem>
				<MenuItem
					to="/items"
					onClick={closeLeftMenu}
				>
					Items
				</MenuItem>
				<MenuItem
					to="/articles"
					onClick={closeLeftMenu}
				>
					Articles
				</MenuItem>
				<MenuItem
					to="/texts"
					onClick={closeLeftMenu}
				>
					Texts
				</MenuItem>
			</div>
		);
	}

	render() {
		const {
			project, leftMenuOpen, closeLeftMenu, userId, dispatchLogout,
			dispatchToggleAuthModal
		} = this.props;

		let isMaincltkProject = false;
		const hostname = getCurrentProjectHostname();

		if (!hostname || ~['cltk.org', 'cltk.local'].indexOf(hostname)) {
			isMaincltkProject = true;
		}

		if (!project) {
			return null;
		}
		return (
			<Drawer
				open={leftMenuOpen}
				docked={false}
				onRequestChange={closeLeftMenu}
				className="leftMenu"
			>
				<LeftMenuHead />
				<div className="leftMenuContent">
					<MenuItem
						to="/"
						onClick={closeLeftMenu}
					>
						Home
					</MenuItem>
					{!isMaincltkProject ?
						this.renderMenuItems()
					: ''}
					<Divider />

					{userId ?
						<div>
							<MenuItem
								to="/profile"
								onClick={closeLeftMenu}
							>
								Profile
							</MenuItem>
							<MenuItem
								to="/profile/projects"
								onClick={closeLeftMenu}
							>
								Projects
							</MenuItem>
							<Divider />

							<MenuItem
								to="/"
								onClick={dispatchLogout}
							>
								Sign out
							</MenuItem>
						</div>
					:
						<MenuItem
							onClick={dispatchToggleAuthModal}
						>
							Sign up / in
						</MenuItem>
					}
				</div>
			</Drawer>

		);
	}
}


LeftMenu.propTypes = {
	project: PropTypes.object,
	leftMenuOpen: PropTypes.bool,
	closeLeftMenu: PropTypes.func,
};

const mapStateToProps = (state, props) => ({
	leftMenuOpen: state.leftMenu.open,
	userId: state.auth.userId,
});

const mapDispatchToProps = (dispatch, ownProps) => ({
	closeLeftMenu: () => {
		dispatch(toggleLeftMenu(false));
	},
	dispatchLogout: () => {
		dispatch(logout(logoutUser));
		dispatch(toggleLeftMenu(false));
	},
	dispatchToggleAuthModal: () => {
		dispatch(toggleAuthModal());
		dispatch(toggleLeftMenu(false));
	},
});


export default connect(
	mapStateToProps,
	mapDispatchToProps,
)(LeftMenu);
