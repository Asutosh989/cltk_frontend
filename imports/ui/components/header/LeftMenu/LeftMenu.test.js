
import React from 'react';
import renderer from 'react-test-renderer';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';

// component:
import LeftMenu from './LeftMenu';

jest.mock('material-ui/List/ListItem');

describe('LeftMenu', () => {
	it('renders correctly', () => {

		const tree = renderer
			.create(
				<MuiThemeProvider>
					<LeftMenu />
				</MuiThemeProvider>
			)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});
});
