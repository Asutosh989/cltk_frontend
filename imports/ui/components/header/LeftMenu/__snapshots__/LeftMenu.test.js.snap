// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeftMenu renders correctly 1`] = `
<div>
  <div
    className="md-sidenav-left"
    style={undefined}
  >
    <div
      className={undefined}
      onTouchTap={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "backgroundColor": "rgba(0, 0, 0, 0.54)",
          "height": "100%",
          "left": "-100%",
          "muiPrepared": true,
          "opacity": 0,
          "pointerEvents": "none",
          "position": "fixed",
          "top": 0,
          "transform": "translateZ(0)",
          "transition": "left 0ms cubic-bezier(0.23, 1, 0.32, 1) 400ms, opacity 400ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": "100%",
          "willChange": "opacity",
          "zIndex": 1200,
        }
      }
    />
    <div
      className={undefined}
      style={
        Object {
          "WebkitOverflowScrolling": "touch",
          "WebkitTapHighlightColor": "rgba(0,0,0,0)",
          "backgroundColor": "#ffffff",
          "borderRadius": "0px",
          "boxShadow": "0 3px 10px rgba(0, 0, 0, 0.16),
                 0 3px 10px rgba(0, 0, 0, 0.23)",
          "boxSizing": "border-box",
          "color": "rgba(0, 0, 0, 0.87)",
          "fontFamily": "Roboto, sans-serif",
          "height": "100%",
          "left": 0,
          "muiPrepared": true,
          "overflow": "auto",
          "position": "fixed",
          "top": 0,
          "transform": "translate(-266px, 0)",
          "transition": "transform 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "width": 256,
          "zIndex": 1300,
        }
      }
    >
      <div
        className="sidenav-top"
      >
        
        <span
          className="user-fullname"
        >
          
        </span>
      </div>
      
      <hr
        style={
          Object {
            "backgroundColor": "#e0e0e0",
            "border": "none",
            "height": 1,
            "margin": 0,
            "marginLeft": 0,
            "marginTop": -1,
            "muiPrepared": true,
          }
        }
      />
    </div>
  </div>
</div>
`;
